src/connect/printer.hpp

```diff
@@ -1,13 +1,15 @@
 #pragma once
 
+#include <transfers/notify_filechange.hpp>
+
 #include <cstdint>
 #include <cstddef>
 #include <tuple>
 #include <optional>
 
 namespace connect_client {
 
-class Printer {
+class Printer : public transfers::NotifyFilechange {
```

Comment:
Do we expect `NotifyFilechange` to be used soon somewhere else too? Adding another inheritance layer only for this one use seems maybe a bit unnecessary  to me :thinking:

Reply:
I'm not sure.

Eventually, we'll want to be able to deliver specific notifications about changes ‒ eg. not only „Something has changed“ but „File of this name was deleted“. This could be a good place to put it and in such case, we would want to use the same mechanism from both Connect and Link.

Nevertheless, the reason I've added this now is that I've needed a „callback“ from the Download:
* I didn't want to use a simple function pointer. Such function would have to call `wui_gcode_modified` directly and it would introduce such dependency into the wrong file ‒ into the planner. The planner is, however, supplied with a printer implementation independent of marlin in tests through the Printer interface. I needed a „this“ pointer / functor object / something like that and passing already existing Printer instance pointer seemed best / most lightweight solution.
* On the other hand, depending on connect::Printer from transfers::Download would create a dependency in the wrong direction ‒ these two directories would then circularly depend on each other. That wouldn't directly break anything, but it „smells“.

So in a sense, this is probably a somewhat underutilizing the potential of the approach a bit, but I got to that in a kind of elimination of other possibilities.

That is, if you have a more elegant solution that doesn't have any of the above unwanted properties, I'm happy to adopt it :-).

Reply:
Ok, you convinced me :D